docs/publishing/index.md

```diff
@@ -0,0 +1,123 @@
+
+In some cases, you may choose to use newer specifications that may require special care when preparing for older browsers. In this case, transpiling down to the level you choose from above may be a good idea. If you choose not to do this, be sure to document this reality thoroughly so that your users are not left unawares.
+
+## Do publish standard es modules
+
+Your package (and those packages that it depends on) should be distributed as standard es modules. This supports the "do" above to deliver the latest standard EcmaScript to your users, modules are both Stage 4 and supported in all major browsers, but also prepares your users to have the simplest time building their applications for production as es modules are the only module format natively supported by browsers.
+
+## Do include appropriate `package.json` entries
+
+Your `package.json` should both its `main` entrypoint and `module` entrypoint to the same es module file. You will find this already configured for you by `npm init @open-wc`, but ensure your `package.json` file includes something simialar to the following:
+
+```
+{
+  "main": "index.js",
+  "module": "index.js",
+  // ...
+}
+```
+
+## Do export element classes
+
+Every custom element is a class extension of `HTMLElement`; e.g. `class MyElement extends HTMLElement {}`. Be sure that you export this class from your package. When starting from `npm init @open-wc` you will see this export in `src/MyElement.js` and reported by the `index.js` entry point. This will make it possible for both customization of your element via extension (e.g. `class NewElement extends MyElement {}`) as well as for advanced usage of the custom element in Scoped Registries, both [native](https://github.com/w3c/webcomponents/issues/716) (when available) or [synthetic](/scoped-elements/).
+
+## Do export side effects separately
+
+A [side effect](<https://en.wikipedia.org/wiki/Side_effect_(computer_science)>) in javascript is created when script effects state outside of the scope in which it is run. This includes when code in an es module effects values available on the `window` or `document` as is inherent to the reality of Custom Elements in that `window.customElements.define()` registers an element in the global registry attached to `window`, whether you explicitly include `window` in your code when doing so or not. JS files with side effects should exist separately from your packages entry point. `npm init @open-wc` exemplifies this through its creation of `my-element.js` which manages the registration of the custom element that this command generates.
+
+## Do keep side effects out of your index.js
+
+When not able to keep code with side effects out of your entry point, be sure to contain such commands within a method (e.g. `run()`, `init()`, etc.) to prevent those side effects from occurring at run time.
```

Comment:
^ This is confusing to me, can we clarify this?

If you're _not_ able to keep side effectful code out of your entrypoint, try to contain it in a method instead? Then its still side effectful, right? It should instead be put in a separate function that has to explicitly be called by the consumer?

Reply:
This is good clarity:
> It should instead be put in a separate function that has to explicitly be called by the consumer?
Was trying to say that...but, you know, fell short...will acquire!